server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/FunctionReferenceEnricher.java

```diff
@@ -28,12 +28,14 @@ public class FunctionReferenceEnricher implements Processor<FunctionReference> {
 
   @Override
   public void accept(FunctionReference node, ProcessingContext processingContext) {
-    SymbolAccumulatorService.FunctionInfo fi = symbolAccumulatorService.getFunctionReference(node.getName());
-    if (fi == null)
-      return;
-
-    node.setDefinitions(fi.getDefinition());
-    node.setUsages(fi.getReferences());
+    node.getProgram()
+        .map(
+            programNode ->
+                symbolAccumulatorService.getFunctionReference(node.getName(), programNode))
```

Comment:
I think this should consider `isFunctionPrefixed`.

Reply:
I removed this. Prefixed function call is taken care by FunctionReferenceProcessor.
Tested on below scenarios
```cobol
       IDENTIFICATION DIVISION.                             
       FUNCTION-ID. func1.                                  
       DATA DIVISION.                                       
       LINKAGE SECTION.                                     
       01  UNRELATED-STUFF.                                 
           05  NUM         PIC X(1234).                     
       PROCEDURE DIVISION RETURNING UNRELATED-STUFF.        
       END FUNCTION func1.    

       IDENTIFICATION DIVISION.                             
       PROGRAM-ID. PGM.                                     
       ENVIRONMENT DIVISION.                                
       CONFIGURATION SECTION.                               
       REPOSITORY.            
               function func1                      
               FUNCTION hex-of intrinsic.                              
       DATA DIVISION.                                        
       WORKING-STORAGE SECTION.                             
       PROCEDURE DIVISION.
       IDENTIFICATION DIVISION.                                
       PROGRAM-ID. PGM1.                                             
       DATA DIVISION.
       WORKING-STORAGE SECTION.  
       01  hex-of pic x.                                 
       PROCEDURE DIVISION.                                   
             display function func1.     *> remove function/remove declaration etc          
       END PROGRAM PGM1.                                                       
       END PROGRAM PGM.
```